src/Akeneo/Bundle/ElasticsearchBundle/Cursor/AbstractCursor.php

```diff
@@ -98,11 +98,34 @@ protected function getNextItems(array $esQuery)
             return [];
         }
 
-        $hydratedItems = $this->repository->getItemsFromIdentifiers($identifiers);
+        $productIdentifiers = array_column(array_filter($identifiers, function ($data) {
+           if (!isset($data['index']))
+           {
+               return true;
+           }
+
+           return $data['index'] === 'pim_catalog_product';
+        }), 'identifier');
+
+        $productModelIdentifiers = array_column(array_filter($identifiers, function ($data) {
+            if (!isset($data['index']))
+            {
+                return true;
+            }
+
+            return $data['index'] === 'pim_catalog_product_model';
+        }), 'identifier');
+
+        $hydratedItems = $this->repository->getItemsFromIdentifiers($productIdentifiers);
+
+        if (!empty($productModelIdentifiers)) {
+            $productModels = $this->productModelRepository->getItemsFromIdentifiers($productModelIdentifiers);
+            $hydratedItems = array_merge($hydratedItems, $productModels);
```

Comment:
Here the cursor handles 2 types of items:
- Product
- ProductModel

We'll need a dedicated cursor which does this work.